File: eforms-gateway/src/main/java/com/bulpros/eformsgateway/form/service/ServiceWithSuppliersStatusEnum.java
package com.bulpros.eformsgateway.form.service;

public enum ServiceWithSuppliersStatusEnum {
    ACTIVE      ("active"),
    PUBLISHED   ("published"),
    INACTIVE    ("inactive"),
    DRAFT       ("draft");

    public final String status;

    private ServiceWithSuppliersStatusEnum(String status) {
        this.status = status;
    }
}

File: eforms-public-portal/src/app/core/types/serviceWithSuppliersStatusEnum.ts
export class serviceWithSuppliersStatusEnum {

  static readonly 'active' = {code: 'active', label: 'SERVICE_WITH_SUPPLIERS.STATUS_ACTIVE'};
  static readonly 'inactive' = {code: 'inactive', label: 'SERVICE_WITH_SUPPLIERS.STATUS_INACTIVE'};
    static readonly 'draft' = {code: 'draft', label: 'SERVICE_WITH_SUPPLIERS.STATUS_DRAFT'};
    static readonly 'published' = {code: 'published', label: 'SERVICE_WITH_SUPPLIERS.PUBLISHED'};

  public static getDisplayByCode(code: string): any {
      let property: string[] = Object.getOwnPropertyNames(serviceWithSuppliersStatusEnum).filter(p => p.toLowerCase() == code.toLowerCase());
      if (property.length > 0) return serviceWithSuppliersStatusEnum[property[0]];
      return undefined;
  }
};